test/base_test_fixture.h

```diff
@@ -16,7 +16,7 @@
 
 #ifdef NANODBC_ENABLE_BOOST
 #include <boost/locale/encoding_utf.hpp>
-#elif defined(__GNUC__) && __GNUC__ < 5
+#elif defined(__GNUC__) && __GNUC__ < 5 && !defined(__llvm__)
```

Comment:
As far as I understand, this is needed for Mac OS regardless of changes to the `CMakeLists.txt`.
If that is correct, then I'd suggest to submit the fix to these conditions in separate PR.

@lexicalunit am I right?

Reply:
It's been a while since I compiled the library but I've only ever worked on macOS myself and I've never needed these. Maybe something changed within the last couple years? 🤷‍♀️

Reply:
No idea, I don't use macOS myself.
Based on @pedro-vicente experience as reported in this issue,  I assume that still is a problem.

Reply:
This PR has 2 issues, I don't know what is the best way to fix, I'll wait for you to make a decision

1) issue 1 is Mac port
as seen in #251 this is the latest MacOS; since you don't have a Mac, I suggest to do whatever changes you feel must be done on this item, test it,  and push here , and I'll test on the Mac

2) issue 2 is CMake

> I think, users who build nanodbc with CMake installed from packages of their OS distributions may complain.

the homebrew installed Cmake is

```
/usr/local/opt/cmake/bin/cmake --version
cmake version 3.18.4
```

so, that should be fine